shell/common/skp_shader_warmup_unittests.cc

```diff
@@ -0,0 +1,237 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <memory>
+
+#include "flutter/assets/directory_asset_bundle.h"
+#include "flutter/flow/layers/container_layer.h"
+#include "flutter/flow/layers/layer.h"
+#include "flutter/flow/layers/physical_shape_layer.h"
+#include "flutter/flow/layers/picture_layer.h"
+#include "flutter/fml/command_line.h"
+#include "flutter/fml/file.h"
+#include "flutter/fml/log_settings.h"
+#include "flutter/fml/unique_fd.h"
+#include "flutter/shell/common/persistent_cache.h"
+#include "flutter/shell/common/serialization_callbacks.h"
+#include "flutter/shell/common/shell_test.h"
+#include "flutter/shell/common/switches.h"
+#include "flutter/shell/version/version.h"
+#include "flutter/testing/testing.h"
+#include "include/core/SkPicture.h"
+#include "include/core/SkSerialProcs.h"
+
+namespace flutter {
+namespace testing {
+
+#if defined(OS_FUCHSIA)
+
+static void WaitForIO(Shell* shell) {
+  std::promise<bool> io_task_finished;
+  shell->GetTaskRunners().GetIOTaskRunner()->PostTask(
+      [&io_task_finished]() { io_task_finished.set_value(true); });
+  io_task_finished.get_future().wait();
+}
+
+class SkpWarmupTest : public ShellTest {
+ public:
+  SkpWarmupTest(){}
+
+  void TestWarmup(const SkISize &draw_size, const LayerTreeBuilder &builder){
+    // Create a temp dir to store the persistent cache
+    fml::ScopedTemporaryDirectory dir;
+    PersistentCache::SetCacheDirectoryPath(dir.path());
+    PersistentCache::ResetCacheForProcess();
+
+    auto settings = CreateSettingsForFixture();
+    settings.cache_sksl = true;
+    settings.dump_skp_on_shader_compilation = true;
+
+    fml::AutoResetWaitableEvent firstFrameLatch;
+    settings.frame_rasterized_callback =
+        [&firstFrameLatch](const FrameTiming& t) { firstFrameLatch.Signal(); };
+
+    auto config = RunConfiguration::InferFromSettings(settings);
+    config.SetEntrypoint("emptyMain");
+    std::unique_ptr<Shell> shell = CreateShell(settings);
+    PlatformViewNotifyCreated(shell.get());
+    RunEngine(shell.get(), std::move(config));
+
+    // Initially, we should have no SkSL cache
+    auto cache = PersistentCache::GetCacheForProcess()->LoadSkSLs();
+    ASSERT_EQ(cache.size(), 0u);
+
+
+    PumpOneFrame(shell.get(), draw_size.width(), draw_size.height(), builder);
+    firstFrameLatch.Wait();
+    WaitForIO(shell.get());
+
+    // Count the number of shaders this builder generated. We use this as a proxy
+    // for whether new shaders were generated, since skia will dump an skp any
```

Comment:
Noted. I think thats ok for the purpose of this test, since the goal is to tell if new shaders were generated and we don't care how many, and in all cases we only pump one frame. I can definitely change the comment to reflect this though